### PR TITLE
[refactor](cv)wait on condition variable more gently

### PR DESCRIPTION
**Pull request by starocean999**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/blocking_queue.hpp [hunk 1]
```diff
@@ -47,25 +48,19 @@ class BlockingQueue {
...
-            if (_shutdown) {
-                return false;
-            }
-
-            timer.start();
-            _get_cv.wait(unique_lock);
-            timer.stop();
+        _get_cv.wait(unique_lock, [this] { return _shutdown || !_list.empty(); });
+        _total_get_wait_time += timer.elapsed_time();
+
```
Comment (yiguolei) [inline be/src/util/blocking_queue.hpp:55]: timer.stop lost? or it is useless?

File: be/src/util/blocking_priority_queue.hpp [hunk 2]
```diff
@@ -49,143 +49,126 @@ class BlockingPriorityQueue {
...
-                    _upgrade_counter = 0;
+        bool wait_successful = false;
+        if (timeout_ms > 0) {
+            wait_successful = _get_cv.wait_for(unique_lock, std::chrono::milliseconds(timeout_ms),
+                                               [this] { return _shutdown || !_queue.empty(); });
+        } else {
+            _get_cv.wait(unique_lock, [this] { return _shutdown || !_queue.empty(); });
+            wait_successful = true;
+        }
+        _total_get_wait_time += timer.elapsed_time();
```
Comment (yiguolei) [inline be/src/util/blocking_priority_queue.hpp:62]: 我感觉这个语义不一致了， 在原来的代码里，即使wait 失败，也会调整队列中已有元素的优先级； 但是在你改的新的代码里，只有wait  成功了，才调整？

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.07 seconds
 load time: 446 seconds
 storage size: 17179983300 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221104113358_clickbench_pr_40025.html

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**